app/src/main/java/com/babylon/wallet/android/presentation/settings/linkedconnectors/LinkedConnectorsScreen.kt

```diff
@@ -228,48 +268,104 @@ private fun ActiveLinkedConnectorsListContent(
 
 @Composable
 private fun ActiveLinkedConnectorContent(
-    activeLinkedConnector: LinkedConnectorsUiState.ConnectorUiItem,
+    activeLinkedConnector: ConnectorUiItem,
     modifier: Modifier = Modifier,
-    onDeleteConnectorClick: (PublicKeyHash) -> Unit,
+    onRenameConnectorClick: (connectorUiItem: ConnectorUiItem) -> Unit,
+    onDeleteConnectorClick: (id: PublicKeyHash) -> Unit,
 ) {
-    Column(modifier = modifier) {
+    Column(modifier = modifier.background(color = RadixTheme.colors.white)) {
         Row(
             Modifier
                 .fillMaxWidth()
                 .padding(horizontal = RadixTheme.dimensions.paddingDefault),
-            horizontalArrangement = Arrangement.spacedBy(RadixTheme.dimensions.paddingSmall),
             verticalAlignment = Alignment.CenterVertically
         ) {
             Text(
                 modifier = Modifier.weight(1f),
                 text = activeLinkedConnector.name,
                 style = RadixTheme.typography.body2Regular,
-                color = RadixTheme.colors.gray2
+                color = RadixTheme.colors.gray2,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis
             )
+            IconButton(
+                onClick = { onRenameConnectorClick(activeLinkedConnector) }
+            ) {
+                Icon(
+                    painter = painterResource(id = com.babylon.wallet.android.designsystem.R.drawable.ic_account_label),
+                    contentDescription = null,
+                    tint = RadixTheme.colors.gray1
+                )
+            }
             IconButton(onClick = {
                 onDeleteConnectorClick(activeLinkedConnector.id)
             }) {
                 Icon(
-                    painter = painterResource(id = R.drawable.ic_delete_24),
+                    painter = painterResource(id = com.babylon.wallet.android.designsystem.R.drawable.ic_delete_outline),
                     contentDescription = null,
                     tint = RadixTheme.colors.gray1
                 )
             }
         }
-        HorizontalDivider(Modifier.fillMaxWidth(), color = RadixTheme.colors.gray4)
     }
 }
 
-@Preview(showBackground = true)
 @Composable
-fun LinkedConnectorsContentWithoutActiveLinkedConnectorsPreview() {
-    RadixWalletTheme {
-        LinkedConnectorsContent(
-            activeLinkedConnectorsList = persistentListOf(),
-            onLinkNewConnectorClick = {},
-            isAddingNewLinkConnectorInProgress = false,
-            onBackClick = {},
-            onDeleteConnectorClick = {}
+private fun RenameActiveLinkedConnectorSheet(
+    input: LinkedConnectorsUiState.RenameConnectorInput,
+    onNewNameChange: (String) -> Unit,
+    onUpdateNameClick: () -> Unit,
+    onDismiss: () -> Unit
+) {
+    val inputFocusRequester = remember { FocusRequester() }
+
+    LaunchedEffect(Unit) {
+        inputFocusRequester.requestFocus()
```

Comment:
Nice! A nice touch is to make the selector show at the end of the current name in the text field, so the user can either remove or edit the current name without the extra tap to set the selector at the right position.

Reply:
I was really looking for exactly this! Any idea how to do it?

Reply:
The only way I know is to set selection to the text field value. As I see we're using `OutlinedTextField`, there is another composable for it that takes `TextFieldValue` type for the `value` param instead of `String`, where you can pass the selection index. Maybe we'd want our `RadixTextField` have a similar overloaded composable which would take a `TextFieldValue` param instead of `String` to be able to handle such cases.

Reply:
I see... thanks for the explanation.
However I see this as a separate task since I played around with the `BasicTextField` which can take a `TextFieldValue` as param, but needs more customization/styling in order to look the same as `RadixTextField`.

Therefore let's better take this in a separate ticket.